src/firefly/js/visualize/HiPSUtil.js

```diff
@@ -0,0 +1,344 @@
+
+import {get} from 'lodash';
+import {clone} from '../util/WebUtil.js';
+import {CysConverter} from './CsysConverter.js';
+import {makeWorldPt, makeDevicePt} from './Point.js';
+import {SpatialVector, HealpixIndex, Utils} from './projection/aladinProj/HealpixIndex.js';
+import {convert, computeDistance} from './VisUtil.js';
+import {replaceHeader} from './WebPlot.js';
+import {primePlot} from './PlotViewUtil.js';
+
+
+/**
+ *
+ * @param {WebPlot} plot
+ * @param {WorldPt} wp new center of projection
+ */
+export function changeProjectionCenter(plot, wp) {
+    if (!plot) return undefined;
+    wp= convert(wp, plot.projection.coordSys);
+    const header= clone(plot.projection.header, {crval1:wp.x, crval2:wp.y});
+    return replaceHeader(plot,header);
+}
+
+
+
+export function getBestHiPSlevelORGIN(plot) {
+    if (!plot) return null;
+    const screenPix= plot.projection.getPixelScaleArcSec()/plot.zoomFactor;
+
+    if (screenPix> 130) return 'allsky';
+
+    const ratio= screenPix / 51;
+    if (ratio> 1/2) return 4;
+    else if (ratio< 1/2) return 5;
+    else if (ratio< 1/4) return 6;
+    else if (ratio<  1/8) return 6;
+    else if (ratio<  1/16) return 7;
+    else if (ratio<  1/32) return 8;
+    else return 9;
+}
```

Comment:
This method is replaced by getBestHiPSlevel, right?

Reply:
yes